src/ray/raylet/scheduling/scheduling_policy_test.cc

```diff
@@ -243,6 +243,41 @@ TEST_F(SchedulingPolicyTest, ForceSpillbackIfAvailableTest) {
   ASSERT_EQ(to_schedule, remote_node);
 }
 
+TEST_F(SchedulingPolicyTest, AvoidSchedulingCPURequestsOnGPUNodes) {
+  // The local node is better, but it has GPUs, the request is
+  // CPU-only, and the remote node does not have GPUs, thus
+  // we should schedule on remote node.
+  StringIdMap map;
+  ResourceRequest req = ResourceMapToResourceRequest(map, {{"CPU", 1}}, false);
```

Comment:
Could we also add test cases with GPU requests, and CPU + non-GPU requests?

Reply:
done